proposals/infra/INF-0004-inproc-outofproc-compiler-library-support.md

```diff
@@ -0,0 +1,138 @@
+<!-- {% raw %} -->
+
+# In-process only or In-process and Out-of-process support for the compiler library
+
+* Proposal: [0004](INF-0004-inproc-outofproc-compiler-library-support.md)
+* Author(s): [Cooper Partin](https://github.com/coopp)
+* Sponsor: [Cooper Partin](https://github.com/coopp)
+* Status: **Under Consideration**
+* Impacted Project(s): (Clang)
+
+* Issues:
+
+## Introduction
+
+An effort is underway to bring compilation support into clang for HLSL based
+shaders.  A C style library will be built to enable applications to compile
+a shader from their own processes in addition to being able to launching the
+clang process.
+
+There are two proposed ways one could consume the shader compiler library and
+compile a shader today.
+
+* **In Process** - An application links against the shader compiler library to
+bring in support for shader compilation as a static library
+(full implementation) or a dynamic library. Compilation is performed within the
+calling application's process.
+
+The DirectX Shader library does not currently support being linked as a
```

Comment:
> The DirectX Shader library does not currently support being linked as a

What does "the DirectX Shader library" refer to?  Is it the same as the "shader compiler library" or the "C style library" mentioned earlier?  Or is this referring to DXC's API?

Reply:
Removed confusion.